packages/now-client/src/check-deployment-status.ts

```diff
@@ -52,87 +45,60 @@ export async function* checkDeploymentStatus(
-
-      for (const build of builds) {
-        const prevState = buildsState[build.id];
-
-        if (!prevState || prevState.readyState !== build.readyState) {
-          debug(
-            `Build state for '${build.entrypoint}' changed to ${build.readyState}`
-          );
-          yield { type: 'build-state-changed', payload: build };
-        }
-
-        if (build.readyState.includes('ERROR')) {
-          debug(`Build '${build.entrypoint}' has errorred`);
-          return yield { type: 'error', payload: build };
-        }
-
-        buildsState[build.id] = build;
-      }
-
-      const readyBuilds = builds.filter((b: DeploymentBuild) => isDone(b));
-
-      if (readyBuilds.length === builds.length) {
-        debug('All builds completed');
-        allBuildsCompleted = true;
-        yield { type: 'all-builds-completed', payload: readyBuilds };
-      }
-    } else {
-      // Deployment polling
-      const deploymentData = await fetch(
-        `${apiDeployments}/${deployment.id || deployment.deploymentId}${
-          teamId ? `?teamId=${teamId}` : ''
-        }`,
-        token,
-        { apiUrl, userAgent }
-      );
-      const deploymentUpdate = await deploymentData.json();
-
-      if (deploymentUpdate.error) {
-        debug('Deployment status check has errorred');
-        return yield { type: 'error', payload: deploymentUpdate.error };
-      }
-
-      if (isReady(deploymentUpdate) && !readyEventFired) {
-        debug('Deployment state changed to READY 2');
-        readyEventFired = true;
-        yield { type: 'ready', payload: deploymentUpdate };
-      }
-
-      if (isAliasAssigned(deploymentUpdate)) {
-        debug('Deployment alias assigned');
-        return yield { type: 'alias-assigned', payload: deploymentUpdate };
-      }
-
-      const aliasError = isAliasError(deploymentUpdate);
-
-      if (isFailed(deploymentUpdate) || aliasError) {
-        debug(
-          aliasError
-            ? 'Alias assignment error has occurred'
-            : 'Deployment has failed'
-        );
-        return yield {
-          type: 'error',
-          payload: aliasError
-            ? deploymentUpdate.aliasError
-            : deploymentUpdate.error || deploymentUpdate,
-        };
-      }
+    // Deployment polling
+    const deploymentData = await fetch(
+      `${apiDeployments}/${deployment.id || deployment.deploymentId}${
+        teamId ? `?teamId=${teamId}` : ''
+      }`,
+      token,
+      { apiUrl, userAgent }
+    );
+    const deploymentUpdate = await deploymentData.json();
+
+    if (deploymentUpdate.error) {
+      debug('Deployment status check has errorred');
+      return yield { type: 'error', payload: deploymentUpdate.error };
+    }
+
+    if (
+      deploymentUpdate.readyState === 'BUILDING' &&
+      !finishedEvents.has('building')
+    ) {
+      debug('Deployment state changed to BUILDING');
+      finishedEvents.add('building');
+      yield { type: 'building', payload: deploymentUpdate };
```

Comment:
Is “type” a string or something more specific? It’s it’s a string we can be more specific like this: https://twitter.com/styfle/status/1208787362613989381

Reply:
There was already a partial type for the `type` property, I've made sure it's a bit more consistent